smt/utils/design_space.py

```diff
@@ -738,17 +743,25 @@ def _correct_get_acting(self, x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
 
         return x_corr, is_acting
 
-    def _sample_valid_x(self, n: int) -> Tuple[np.ndarray, np.ndarray]:
+    def _sample_valid_x(self, n: int, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
         """Sample design vectors"""
 
         # Simplified implementation: sample design vectors in unfolded space
         x_limits_unfolded = self.get_unfolded_num_bounds()
-        sampler = LHS(xlimits=x_limits_unfolded, random_state=self.seed)
-        x = sampler(n)
-
-        # Cast to discrete and fold
-        self._normalize_x(x)
+        if "random_state" in kwargs:
+            self.seed = kwargs["random_state"]
+        if kwargs.get("use_new_sampler"):
+            kwargs.pop("use_new_sampler", None)
+            if self.use_new_sampler:
```

Comment:
The logic here is a bit convoluted. Why do you need `self.use_new_sampler`? Another point for testing properly `use_new_sampler` in isolation in `DesignSpace` tests. ;)